crates/uv/src/commands/pip/tree.rs

```diff
@@ -82,18 +89,18 @@ pub(crate) fn pip_tree(
 // For example, `requests==2.32.3` requires `charset-normalizer`, `idna`, `urllib`, and `certifi` at
 // all times, `PySocks` on `socks` extra and `chardet` on `use_chardet_on_py3` extra.
 // This function will return `["charset-normalizer", "idna", "urllib", "certifi"]` for `requests`.
-fn required_with_no_extra(dist: &InstalledDist) -> Vec<pep508_rs::Requirement<VerbatimParsedUrl>> {
+fn required_with_no_extra(
+    dist: &InstalledDist,
+    marker_environment: &MarkerEnvironment,
+) -> Vec<pep508_rs::Requirement<VerbatimParsedUrl>> {
     let metadata = dist.metadata().unwrap();
     return metadata
         .requires_dist
         .into_iter()
         .filter(|r| {
             r.marker.is_none()
-                || !r
-                    .marker
-                    .as_ref()
-                    .unwrap()
-                    .evaluate_optional_environment(None, &metadata.provides_extras[..])
+                || (r.marker.as_ref().unwrap().evaluate(marker_environment, &[])
+                    && !r.marker.as_ref().unwrap().has_extra())
```

Comment:
Is this not sufficient?

```rust
requirement.marker
    .as_ref()
    .map_or(true, |m| m.evaluate(marker_environment, &[]))
```

Why do we need the `has_extra` piece?

Reply:
oops, you're right, removed 👍